src/worker/plugins/setup.ts

```diff
@@ -24,6 +25,10 @@ export async function setupPlugins(server: PluginsServer): Promise<void> {
         } else {
             pluginConfig.vm = new LazyPluginVM()
             pluginVMLoadPromises.push(loadPlugin(server, pluginConfig))
+
+            if (prevConfig && (await prevConfig?.vm?.getTeardownPlugin())) {
```

Comment:
I think it'd be alright to avoid calling `getTeardownPlugin()` since `teardownPlugins` already does it and skips as needed?

Reply:
Makes sense. Removed the check from here, as it was also somehow leaking abstractions (details from within `teardownPlugins` made their way out into `setupPlugins`...)